### DISCUSSION TITLE
Home Assistant ed indirizzi IP

### DISCUSSION THREAD

**Discussion by alessiomanzi** [indomus/forum#2031] [HUB: Home Assistant]
Salve, e perdonate la mia poca esperienza nei sistemi di rete, leggo che molti utenti hanno oltre 500 dispositivi configurati. Mi chiedo, se ho una rete con indirizzi IP da 0 a 254 , come posso superare questo limite in modo che HA li possa vedere tutti in trasparenza e magari assegnando ad essi un IP statico?
Grazie anticipatamente a chi mi schiarirà il dubbio.

**Comment by snakuzzo** [comment]
Di che limite parli ?
Sei sulla stessa rete e di base si vedono tutti.
Se ti serve che i device abbiano sempre lo stesso IP, su tutti i router in commercio esiste la possibilità di assegnare gli IP fissi in base ai mac address.

**Reply by alessiomanzi** [reply]
Grazie della risposta , intendevo quando si esaurisce l'indirizzamento dei 254 indirizzi IP , ora so che generando una subnet gli indirizzi possono diventare 254X254 ma HA , il router e la macchina virtuale che ospita l'istanza di HA, come devono essere configurati ?

**Reply by snakuzzo** [reply]
Sinceramente... il fatto che "molti utenti hanno oltre 500 dispositivi configurati" mi sembra sia molto strano, in tutti questi anni non ho mai letto di persone che avevano tutta questa roba :)
Non sono un esperto (anz), ma ci possono essere diverse soluzioni e dipende anche da cosa ti permette il tuo router.
Solitamente bastano gli IP inclusi nella subnet di default che è una `/24`  (es. 192.168.1.0/24), ma se vuoi ampliare potresti configurare sul router una `/23`... e così via a seconda delle esigenze (a tal proposito dai un'occhiata [QUI](https://www.ipaddressguide.com/cidr))
Se non sai di cosa parlo, dovresti studiarti come funzionano le subnet...non è banale. Per cui, se non sei pratico...ti consiglierei di lasciar perdere. Credo sia molto raro arrivare ad oltre 200 device connessi...considerando anche che non tutti i router ti permettono di gestirne così tanti.

**Comment by alessiomanzi** [comment]
Grazie , 
il fatto che si debba studiare non è un problema , non sono un sistemista come avrai capito , ma vengo dal mondo dell'impiantistica e dei PLC dove tutto funziona in modo diverso e con costi esorbitanti. In HA i concetti da sapere volgono molto di più sui sistemi ed i linguaggi , cosa che senz'altro studierò.

**Comment by KC-inDomus** [comment]
> Salve, e perdonate la mia poca esperienza nei sistemi di rete, leggo che molti utenti hanno oltre 500 dispositivi configurati. Mi chiedo, se ho una rete con indirizzi IP da 0 a 254 , come posso superare questo limite in modo che HA li possa vedere tutti in trasparenza e magari assegnando ad essi un IP statico? Grazie anticipatamente a chi mi schiarirà il dubbio.

se come la stragrande maggioranza delle reti lan hai una subnet /24, effettivamente al massimo hai 254 host utili. ma già se usi una /22 nel hai 1022. comunque, 500 dispositivi configurati, trovami un utente uno che li abbia, in casa :) perché tutti questi problemi?

**Comment by depahk** [comment]
@alessiomanzi Mai letto di utenti con 500, ne 200 ip attivi in rete. Io a casa ho più reti (/24) e ospito server, vm, iot, cam, ecc.. ma supero solo il centinaio di IP e dubito fortemente che tu possa arrivare a saturare una /24. Ti direi di dormire sonni tranquilli, se mai si presenterà il problema in futuro avrai acquisito le nozioni che ti servono per sistemare la rete prima perché gestire 250 dispositivi non è una cosa semplice.
